src/Service/QueueVideoImports.php

```diff
@@ -0,0 +1,174 @@
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Queues MpxImportTask items to be processed by background processes.
+ *
+ * @package Drupal\media_mpx\Service
+ */
+class QueueVideoImports {
+
+  const MEDIA_MPX_IMPORT_QUEUE = 'media_mpx_importer';
+
+  /**
+   * The Logger channel.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  private $logger;
+
+  /**
+   * The Event Dispatcher.
+   *
+   * @var \Symfony\Component\EventDispatcher\EventDispatcherInterface
+   */
+  private $eventDispatcher;
+
+  /**
+   * The Data Object Factory Creator.
+   *
+   * @var \Drupal\media_mpx\DataObjectFactoryCreator
+   */
+  private $dataObjectFactoryCreator;
+
+  /**
+   * The mpx Media Types repository.
+   *
+   * @var \Drupal\media_mpx\Repository\MpxMediaType
+   */
+  private $mpxMediaTypeRepository;
+
+  /**
+   * The queue to use for mpx video imports.
+   *
+   * @var \Drupal\Core\Queue\QueueInterface
+   */
+  private $queue;
+
+  /**
+   * QueueContentsForm constructor.
+   */
+  public function __construct(MpxMediaType $mpxMediaTypeRepository, LoggerChannelInterface $loggerChannel, EventDispatcherInterface $eventDispatcher, DataObjectFactoryCreator $dataObjectFactoryCreator, QueueFactory $queueFactory) {
+    $this->logger = $loggerChannel;
+    $this->eventDispatcher = $eventDispatcher;
+    $this->dataObjectFactoryCreator = $dataObjectFactoryCreator;
+    $this->queue = $queueFactory->get(self::MEDIA_MPX_IMPORT_QUEUE);
+    $this->mpxMediaTypeRepository = $mpxMediaTypeRepository;
+  }
+
+  /**
+   * Executes the queue service with the given parameters.
+   *
+   * @param \Drupal\media_mpx\Service\QueueVideoImportsRequest $request
+   *   The request object containing the media type and query filters.
+   *
+   * @return \Drupal\media_mpx\Service\QueueVideoImportsResponse
+   *   A response object with data about the queued videos.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\media_mpx\Exception\MediaTypeDoesNotExistException
+   * @throws \Drupal\media_mpx\Exception\MediaTypeNotAssociatedWithMpxException
+   */
+  public function execute(QueueVideoImportsRequest $request): QueueVideoImportsResponse {
+    $media_type_id = $request->getMediaTypeId();
+    $limit = $request->getLimit();
+
+    $media_type = $this->mpxMediaTypeRepository->findByTypeId($media_type_id);
+
+    $results = $this->fetchMediaTypeItemIdsFromMpx($media_type, $request);
+    $queued = 0;
+    $errored = 0;
+    foreach ($results as $index => $mpx_media) {
+      if (!is_null($limit) && ($queued + $errored) >= $limit) {
+        break;
+      }
+
+      if ($this->queueMpxItem($mpx_media, $media_type->id()) == TRUE) {
+        $queued++;
+      }
+      else {
+        $errored++;
+      }
+    }
+
+    return new QueueVideoImportsResponse($queued, $errored, $results);
```

Comment:
I see a few ways to reduce the complexity of this function, but I actually think this class is the root of things.

The problem with this class is that we're storing _counts_ of `$queued` and `$errored`, but losing track of which specific mpx items passed or failed. As a site admin, if something fails I generally want to know which video failed so I can try to fix it. If we were not sending in `$results` this would make sense, because we could potentially let the GC free those mpx media objects, but we're still keeping a reference around.

If `queueMpxItem` returns an object, we could have code that looks something like:

```php
$limit = min($limit, count($results));
for ($index = 0; $index < $limit; $index++) {
  $queue_results[] = $this->queueMpxItem($mpx_media, $media_type->id());
}
return new QueueVideoImportsResponse($queue_results);
```

I don't think we need to hold this PR on this, but let's get a followup done before closing our ticket.